src/hooks/useProtocolManager.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react"
+import { useContract, useProvider, useSigner } from "wagmi"
+import { SherlockProtocolManager } from "../contracts"
+import SherlockProtocolManagerABI from "../abi/SherlockProtocolManager.json"
+import { BigNumber, ethers } from "ethers"
+
+/**
+ * Address of Sherlock Protocol Manager contract
+ */
+export const SHERLOCK_PROTOCOL_MANAGER_ADDRESS = process.env.REACT_APP_SHERLOCK_PROTOCOL_MANAGER_ADDRESS as string
+
+/**
+ * Array of protocols covered by Sherlock
+ */
+export const COVERED_PROTOCOLS = {
+  SQUEETH: {
+    name: "Squeeth by Opyn",
+  },
+  EULER: {
+    name: "Euler",
+  },
+  PRIMITIVE: {
+    name: "Primitive",
+  },
+  NIFTY_OPTIONS: {
+    name: "Nifty Options by Teller",
+  },
+}
+
+/**
+ * React Hook for interacting with Sherlock Protocol Manager contract.
+ *
+ * See https://github.com/sherlock-protocol/sherlock-v2-core
+ */
+const useProtocolManager = () => {
+  const provider = useProvider()
+  const [{ data: signerData }] = useSigner()
+  const contract: SherlockProtocolManager = useContract({
+    addressOrName: SHERLOCK_PROTOCOL_MANAGER_ADDRESS,
+    signerOrProvider: signerData || provider,
+    contractInterface: SherlockProtocolManagerABI.abi,
+  })
+
+  /**
+   * Fetch a protocol's active balance.
+   *
+   * See https://docs.sherlock.xyz/protocols/premiums#maintaining-an-active-balance
+   */
+  const getProtocolActiveBalance = React.useCallback(
+    async (protocol: keyof typeof COVERED_PROTOCOLS): Promise<BigNumber> => {
+      return contract.activeBalance(ethers.utils.formatBytes32String(protocol))
+    },
+    [contract]
+  )
```

Comment:
I've been returning the entire Contract object and make the component call the inner functions, but maybe this is a better approach and we only expose what we need.

I'd be much better for unit testing, too.